0.data_analysis_and_processing/nbconverted/filter_nuclei.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Filter Single-Cell Profiles
+# Single-cell profiles are filtered by nuclei bounding box dimensions across all plates.
+# Filtered is accomplished with robust z-score thresholding of nuclei bounding box dimensions.
+
+# In[1]:
+
+
+import pathlib
+import warnings
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import seaborn as sns
+
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+# Remove categorical warning in seaborn
+warnings.filterwarnings("ignore", category=FutureWarning)
```

Comment:
I haven't seen this before. Just for my own curiosity, what is this warning?

Reply:
This is for a warning that shows when I try to create the plots. If you know why I get this warning, I would be interested to know:
`/home/user/mambaforge-pypy3/envs/speckle_analysis/lib/python3.11/site
-packages/seaborn/_oldcore.py:1498: FutureWarning: is_categorical_dty
pe is deprecated and will be removed in a future version. Use isinsta
nce(dtype, CategoricalDtype) instead`